### PR TITLE
update workflows with current runners and up to date actions

### PR DESCRIPTION
**Pull request by jgreat**
The workflow configs need updating to use current runners and more up to date actions.

Sorry that this one is messy, there was clean up to make spacing and indenting consistent.

- Move to new runner configs.
- Update workflows to use mobilecoin actions where appropriate.

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -21,53 +21,54 @@ permissions:
...
+    - name: Checkout
+      uses: mobilecoinofficial/gh-actions/checkout@v0
+
+    - name: Setup Rust
+      uses: ./.github/actions/setup-rust
+
+    - name: Cargo test
+      shell: bash
+      run: |
+        cargo test
```
Comment (nick-mobilecoin) [inline .github/workflows/CI.yml]: This lost the `--locked` flag. I don't think it matters as i think the build will catch it even for tests
```suggestion
        cargo test --locked

File: .github/workflows/build-deploy-staging.yaml [hunk 2]
```diff
@@ -15,146 +15,109 @@ on:
...
+        release_name: ${{ matrix.target.chain }}net-${{ env.REPO_NAME }}
+        rancher_url: ${{ secrets.RANCHER_B_URL }}
+        rancher_token: ${{ secrets.RANCHER_B_TOKEN }}
+        chart_repo: ${{ env.CHART_REPO }}
+        chart_name: ${{ env.REPO_NAME}}
+        chart_version: ${{ github.ref_name }}
+        chart_values: ${{ env.CHART_PATH }}/${{ matrix.target.chain }}net-values.yaml
+        # CBB: change mobilecoind.image to reserveAuditor.image...
+        chart_set: |
+          --set=mobilecoind.image.tag=${{ github.ref_name }}-${{ matrix.target.chain }}net
```
Comment (nick-mobilecoin) [inline .github/workflows/build-deploy-staging.yaml:121]: is it ok that `--set=frontend.image.tag....` went away?

Comment (nick-mobilecoin) [reply .github/workflows/build-deploy-staging.yaml:121]: I see it also went away in `dispatch-deploy.yaml`

**Review by nick-mobilecoin — APPROVED** [review]

File: .github/workflows/build-deploy-staging.yaml (hunk 2 above)
Comment (jgreat) [reply .github/workflows/build-deploy-staging.yaml:121]: Should be. The `chart_version` sets the default for the frontend image tag, no need so specify it again.

**Pull request merged**